library/src/main/java/bio/terra/landingzone/stairway/flight/exception/utils/ManagementExceptionUtils.java

```diff
@@ -0,0 +1,32 @@
+package bio.terra.landingzone.stairway.flight.exception.utils;
+
+import com.azure.core.management.exception.ManagementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ManagementExceptionUtils {
+  private ManagementExceptionUtils() {}
+
+  public static String buildErrorInfo(ManagementException e) {
+    StringBuilder errorDetails = new StringBuilder();
+    errorDetails.append("ErrorMessage: %s; ".formatted(e.getMessage()));
+    if (e.getValue() != null) {
+      errorDetails.append(
+          "ErrorCode: %s; AdditionalMessage: %s;"
+              .formatted(
+                  Optional.of(e.getValue().getCode()).orElse("n/a"),
+                  Optional.of(e.getValue().getMessage()).orElse("n/a")));
+      if (e.getValue().getDetails() != null) {
+        errorDetails.append(
+            "Details: %s;"
+                .formatted(
+                    e.getValue().getDetails().stream()
+                        .map(
+                            me ->
+                                "[code: %s, message: %s]".formatted(me.getCode(), me.getMessage()))
+                        .collect(Collectors.joining(","))));
```

Comment:
Do you have an example of what this ends up looking like in practice?

Reply:
I don't have an example, but based on the exception I provided above in the PR's description it might look like:

ErrorMessage: Long Running operation Failed or Cancelled; ErrorCode: ResourceOperationFailure; AdditionalMessage: The resource operation completed with terminal provisioning state 'Failed'.; Details: [code: InternalServerError, message: An unexpected error occured while processing the request. Tracking ID: '8a04238e-7145-42d6-a71e-8a82d6f9f819'];

Details represents a list of different issues.